### DISCUSSION TITLE
Frequency Tolerances and the SX127X (was SF 12 issues with SX1276 at 125k BW)

### DISCUSSION THREAD

**Discussion by faragher** [markqvist/Reticulum#384] [General]
As part of a larger project, I did some digging on SX127X tolerances. Because this is highly useful troubleshooting information, I'm going to post it here immediately.

Buried in the SX127X datasheet (seriously, I was chasing down terms that are only half searchable until I finally found what I needed) are actual frequency tolerances for transmission. Bear in mind this is total difference between sender and receiver, so it's possible two in-spec receivers can be out of spec to each other. 

Tolerance is 25% of the bandwidth or 50/100/200 ppm for SF 12/11/10 respectively, whichever is lower. I have prepared this quick reference table (915 MHz, 868 MHz will be slightly more taxing, and 433 will require more than twice the precision):

![FreqTol](https://github.com/markqvist/Reticulum/assets/47290019/4ba18dfb-621c-44a7-9d45-d8bb90a3ed87)

The LilyGO units I use do not have an exposed chipset, and the schematics don't list a tolerance or any sub-component information within their "TTGO LORA" integrated circuit package. If I'm reading the SX127X datasheet correctly, it appears to have an onboard 20ppm XTAL, which as noted in the link below, may stray from this precision with changes in temperature.

Without a change to a TCXO (around 1ppm) this puts a general limit on usable settings. The datasheet suggests all bandwidths under 62.5kHz being used with TXCOs, which is already pushing the edge of 20ppm tolerance. While it would take a number of things going wrong to hit the rated 34ppm for SF12 BW125, it's feasible.

Anyway, just figured it might be helpful.

Original message follows:

I know we've had some iffy results with SF12 sometimes, and it looks like we're not the only ones.

https://www.rocketscream.com/blog/2017/08/21/the-sx1276-modules-shootout-hoperfs-rfm95w-vs-nicerfs-lora1276-c1-vs-hpdteks-hpd13/

It looks like their oscillators may not have a tight enough tolerance for those settings under certain temperature conditions. I recommend not using SF12 for production networks at this time. It's not worth possible connectivity problems with end-user hardware.  It's hard enough troubleshooting without hardware variations being a problem. 

This is just me remembering SF12 problems and an article I found while researching HopeRF RFM95 compatibility. But it's worth sharing that it is a known issue.

**Comment by cobraPA** [comment]
Nice, thanks for the info.
